tensilelite/Tensile/Source/lib/source/ContractionSolution.cpp

```diff
@@ -775,36 +775,46 @@ namespace Tensile
         }
 
         uint32_t gsu = param.gsu() > 0 ? param.gsu() : sizeMapping.globalSplitU;
-        uint32_t wgm = param.wgm() > 0 ? param.wgm() : sizeMapping.workGroupMapping;
-        if(!internalArgsSupport.wgm)
-            wgm = 0;
-
-        const uint32_t mask         = 0xFF;
-        uint32_t       internalArg0 = mask & gsu;
-        uint32_t       wgShift8     = (mask & wgm) << 8;
-        internalArg0                = internalArg0 | wgShift8;
+        int32_t wgm  = param.wgm() != 0 ? param.wgm() : sizeMapping.workGroupMapping;
+
+        const uint32_t mask16       = 0xFFFF;
+        const uint32_t mask8        = 0xFF;
+        uint32_t       internalArg0 = mask16 & gsu;
```

Comment:
Strictly speaking, this should be mask8 in customKernel case.
(it might not be an issue unless we use GSU>255)

Reply:
Will change